samples/nextjs/src/lib/with-personalization-and-tracking.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  isServer,
+  LayoutPersonalizationService,
+  withSitecorePersonalizationContext,
+  areQueryParamsReady,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import { TrackingService } from '../../../../packages/sitecore-jss-tracking/types';
+import { SitecorePageProps } from 'lib/page-props';
+import { useRouter } from 'next/router';
+
+export function withCurrentPageTracking<T extends SitecorePageProps>(
+  Component: React.ComponentClass<T> | React.SFC<T>,
+  trackingService: TrackingService
+): (props: T) => JSX.Element {
+  return function WithCurrentPageTrackingHoc(props: T) {
+    const newProps = { ...props };
+    const route = props.layoutData?.sitecore.route;
+    if (
+      !props.isPreview &&
+      !props.tracked &&
+      route &&
+      route.layoutId !== 'available-in-connected-mode' &&
+      areQueryParamsReady(useRouter()) &&
+      !isServer()
+    ) {
+      trackingService.trackCurrentPage(props.layoutData?.sitecore.context, route);
+      newProps.tracked = true;
+    }
+    return <Component {...newProps} />;
+  };
+}
+
+export function withNotFoundPageTracking<T>(
```

Comment:
This signature doesn't fit with our HOC's signature.
Use: `withNotFoundPageTracking(options)(Component)`